wgpu-core/src/pipeline.rs

```diff
@@ -299,8 +299,8 @@ pub enum ColorStateError {
     FormatNotBlendable(wgt::TextureFormat),
     #[error("Format {0:?} does not have a color aspect")]
     FormatNotColor(wgt::TextureFormat),
-    #[error("Format {0:?} can't be multisampled")]
-    FormatNotMultisampled(wgt::TextureFormat),
+    #[error("Sample count {0} is not supported by format {1:?} on this device. It may be supported by your adapter through the TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES feature.")]
```

Comment:
```suggestion
    #[error("Sample count {0} is not supported by format {1:?} on this device. The WebGPU spec guarentees {2:?} samples are supported on this format. With the TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES feature, your device supports {3:?}.")]
```

Adding the lists of supported sample counts will make this a lot more actionable. The error case is a slow path, so don't worry about not writing the most efficient struct.

Reply:
I think we would have to instantiate a new adapter with the TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES feature in order to know what sample counts are supported by that feature. Instantiating a new adapter seems like too much for an error message.

I did improve the error message further though, take a look.

Reply:
The feature is just an internal logic gate, you should be able to (where these errors are made) call `adapter.get_texture_format_features` even with the feature off.

Reply:
I don't know how to get guaranteed format features? I've seen the function `guaranteed_format_features`, but it requires the device features which I don't have access to at the points in the code where these errors are created.

Reply:
I did manage to produce error messages such as:

```
Sample count 8 is not supported by format Bgra8UnormSrgb on this device (supported sample counts are: [1, 4]). The TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES feature enables support for sample counts [1, 2, 4, 8].
```

and

```
Sample count 3 is not supported by format Bgra8UnormSrgb on this device (supported sample counts are: [1, 2, 4, 8]). The TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES feature enables support for sample counts [1, 2, 4, 8].
```

That last one is a little confusing though.

Reply:
The features are available from the device - in this case `self.features`.

Reply:
I believe this is implemented the way you wanted with the wording you wanted. Thanks for the help.